### ISSUE TITLE
[Feature]: Add `setLastKnownLocation()` to the SDK

### ISSUE THREAD

**Issue by PhillipFraThailand** [braze-inc/braze-react-native-sdk#135]
Hi guys, I am implementing Braze into an application and i am using the [Web JS documentation](https://js.appboycdn.com/web-sdk/latest/doc/modules/appboy.html) as a reference.

I need to store the last known latitude and longitude of the user, and was hoping to use the method from the documentation called `setLastKnownLocation()`. But it looks like this method isn't supported in the React Native SDK, and same goes for `getUserId()`.

I found the method `setCustomLocationAttribute()` and implemented it like this:
```
export const addAttributesToBrazeUser = (country: string, language: string, latitude: number, longitude: number) => {
  ReactAppboy.setLocationCustomAttribute('testsetLocationCustomAttribute', latitude, longitude)
};
```

But i haven't been able to see the effect of the function, and i dont know if it is an alternative solution. 

I would also like to know if there is any more elaborate documentation somewhere or any other good places to look?

Thanks in advance :)

**Comment by wesleyorbin** [comment]
Hi @PhillipFraThailand. Thanks for reaching out. You're correct that `setLastKnownLocation()` does not exist on the React SDK. `setCustomLocationAttribute()` does not have the same effect as `setLastKnownLocation()`. We've filed an internal ticket to add it to the SDK.

Can you elaborate on your use case for `getUserId()` on the React SDK?

**Comment by PhillipFraThailand** [comment]
Hey Wesley, thanks for the Clarification. Do you know if there are any good ressources for clarification on terms etc? I feel like the current web documentation requires just a tiny bit too much domain knowledge for a person to start getting comfortable with implementing Braze in a good pace. Also can you give me some kind of timeframe of when the functionality   of `setLastKnownocation()` could be implemented?

For `getUserId()` i want this to be able to easily debug for example, being able to see exactly what the userId the current session has. Also if we successfully connected to the correct user across devices. We are still just getting started with Braze, but i could imagine that we would also need it to conditionally update/set attributes and tracking events, based on whether the user is 'logged in' already, since we have an app that is usable also while not logged in.

**Comment by wesleyorbin** [comment]
@PhillipFraThailand Thank you for the feedback and elaboration on your use case. The [Braze developer guide](https://www.braze.com/docs/developer_guide/home/) is the best place to start, as it explains our various features and terminology more than the class documentation. 

I've also created an internal ticket to add `getUserId()` to the React SDK. We don't have a specific timeframe for `setLastKnownLocation()` right now, but we should be able to get it into the next release.

**Comment by PhillipFraThailand** [comment]
Thank you @wesleyorbin. We are looking forward to the next release 😊

**Comment by GrandjeanIo** [comment]
@wesleyorbin any ETA as to when the method will be available?

**Comment by hokstuff** [comment]
Hi @PhillipFraThailand  @GrandjeanIo,

We have just released [React Native SDK version 7.0.0](https://github.com/braze-inc/braze-react-native-sdk/releases/tag/7.0.0) which adds `Braze.setLastKnownLocation()` to the SDK.

Thanks for your patience!
